scipy/stats/_rvs_sampling.py

```diff
@@ -168,3 +171,451 @@ def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
+
+    >>> import numpy as np
+    >>> from scipy import stats
+    >>> from timeit import timeit
+    >>> time_once = lambda f: f"{timeit(f, number=1)*1000:.6} ms"
+    >>> dist = stats.genexpon(9, 16, 3)  # freeze the distribution
+    >>> p = np.linspace(0.01, 0.99, 99)  # percentiles from 1% to 99%
+    >>> time_once(lambda: dist.ppf(p))
+    '154.565 ms'  # may vary
+
+    >>> time_once(lambda: dist.rvs(size=100))
+    '148.979 ms'  # may vary
+
+    The `NumericalInverseHermite` has a method that approximates ``dist.ppf``.
+
+    >>> from scipy.stats import NumericalInverseHermite
+    >>> fni = NumericalInverseHermite(dist)
+    >>> np.allclose(fni.ppf(p), dist.ppf(p))
+    True
+
+    In some cases, it is faster to both generate the fast numerical inverse
+    and use it than to call ``dist.ppf``.
+
+    >>> def time_me():
+    ...     fni = NumericalInverseHermite(dist)
+    ...     fni.ppf(p)
+    >>> time_once(time_me)
+    '11.9222 ms'  # may vary
+
+    After generating the fast numerical inverse, subsequent calls to its
+    methods are much faster.
+    >>> time_once(lambda: fni.ppf(p))
+    '0.0819 ms'  # may vary
+
+    The fast numerical inverse can also be used to generate random variates
+    using inverse transform sampling.
+
+    >>> time_once(lambda: fni.rvs(size=100))
+    '0.0911 ms'  # may vary
+
+    Depending on the implementation of the distribution's random sampling
+    method, the random variates generated may be nearly identical, given
+    the same random state.
+
+    >>> # `seed` ensures identical random streams are used by each `rvs` method
+    >>> seed = 500072020
+    >>> rvs1 = dist.rvs(size=100, random_state=np.random.default_rng(seed))
+    >>> rvs2 = fni.rvs(size=100, random_state=np.random.default_rng(seed))
+    >>> np.allclose(rvs1, rvs2)
+    True
+
+    To use `NumericalInverseHermite` with a custom distribution, users may
+    subclass  `scipy.stats.rv_continuous` and initialize a frozen instance or
+    create an object with equivalent ``pdf``, ``cdf``, and ``ppf`` methods.
+    For instance, the following object represents the standard normal
+    distribution. For simplicity, we use `scipy.special.ndtr` and
+    `scipy.special.ndtri` to compute the ``cdf`` and ``ppf``, respectively.
+
+    >>> from scipy.special import ndtr, ndtri
+    >>>
+    >>> class MyNormal:
+    ...
+    ...     def pdf(self, x):
+    ...        return 1/np.sqrt(2*np.pi) * np.exp(-x**2 / 2)
+    ...
+    ...     def cdf(self, x):
+    ...        return ndtr(x)
+    ...
+    ...     def ppf(self, x):
+    ...        return ndtri(x)
+    ...
+    >>> dist1 = MyNormal()
+    >>> fni1 = NumericalInverseHermite(dist1)
+    >>>
+    >>> dist2 = stats.norm()
+    >>> fni2 = NumericalInverseHermite(dist2)
+    >>>
+    >>> print(fni1.rvs(random_state=seed), fni2.rvs(random_state=seed))
+    -1.9603810921759424 -1.9603810921747074
+
+    """
+
+    def __init__(self, dist, tol=1e-12, max_intervals=100000):
```

Comment:
```suggestion
    def __init__(self, dist, *, tol=1e-12, max_intervals=100000):
```
To allow more flexibility in developing the interface. The keyword only requirement can be removed later, if desired, but it is possible we will want to add new arguments immediately after `dist`, so makes sense to set the requirement in this initial release.